Dockerfile

```
@@ -24,17 +26,17 @@ RUN cd contracts && npm install
 # Switch back to root to install the Go binary
 USER root
 
-# Build the Go binary in a separate stage
-FROM golang:1.22 as builder
+# Build the Go binary in a separate stage utilizing Makefile
+FROM golang:1.22 AS builder
 WORKDIR /app
 COPY go.mod go.sum ./
 RUN go mod download
 COPY . .
```

mmmh only last thing. we might need to copy actually `.git` manually because `COPY .` it probably won't copy it over
```suggestion
COPY .git .
COPY . .
```

---

I just ran it and it seems to be fine but I'll double check.